cloud/satellite/satellite_activation_keys.py

```diff
@@ -0,0 +1,365 @@
+    resp_json = json.loads(resp.read())
+
+    if resp_json["total"] == 0:
+        module.fail_json(msg="Environment %s not found." % env_name)
+    elif resp_json["total"] > 1:
+        module.fail_json(msg="Environment name %s not specific enough, more than one found." % env_name)
+    env_id = resp_json["results"][0]["id"]
+
+    return env_id
+
+def get_organization_id(module):
+    """Looks up organization ID for a given organization name.
+    """
+
+    org_name = module.params.get('organization')
+
+    url_action = 'https://' + module.params.get('satellite') + '/katello/api/v2/organizations/'
+    payload = {"name":org_name}
+    headers = {'Content-Type':'application/json'}
+
+    resp = open_url(url_action,method="GET",headers=headers,url_username=module.params.get('user'),url_password=module.params.get('pwd'),force_basic_auth=True,data=json.dumps(payload),validate_certs=module.params.get('validate_certs'))
+    resp_json = json.loads(resp.read())
+
+    if resp_json["total"] == 0:
+        module.fail_json(msg="Organization %s not found." % org_name)
+    elif resp_json["total"] > 1:
+        module.fail_json(msg="Organization name %s not specific enough, more than one found." % org_name)
+    orga_id = resp_json["results"][0]["id"]
+
+    return orga_id
+
+def get_key(module,orga_id):
+    """Looks up entire key for a given key name. The entire key will be returned.
+    """
+
+    key_name = module.params.get('key_name')
+
+    url_action = 'https://' + module.params.get('satellite') + '/katello/api/v2/organizations/' + str(orga_id) + '/activation_keys/?search'
+    payload = {"name":key_name}
+    headers = {'Content-Type':'application/json'}
+
+    resp = open_url(url_action,method="GET",headers=headers,url_username=module.params.get('user'),url_password=module.params.get('pwd'),force_basic_auth=True,data=json.dumps(payload),validate_certs=module.params.get('validate_certs'))
+    resp_json = json.loads(resp.read())
+
+    try:
+        resp_json["total"]
+    except KeyError:
+        return None
+
+    if resp_json["total"] == 0:
+        return None
+    elif resp_json["total"] > 1:
+        module.fail_json(msg="Key name %s not specific enough, more than one found." % key_name)
+
+    key = resp_json["results"][0]
+
+    return key
+
+def manage_key(module):
+    """Retrieves all module information and performs key management REST calls. Returns changed state and information.
+    """
+
+    user      = module.params.get('user')
+    pwd       = module.params.get('pwd')
+    satellite = module.params.get('satellite')
+    organization      = module.params.get('organization')
+    key_action  = module.params.get('key_action')
+    key_name  = module.params.get('key_name')
+    validate_certs = module.params.get('validate_certs')
```

Comment:
@woot4moo I am still not sure where to introduce kwargs in the best way: the manage_key function still needs to have the entire module, right? Or would you already hand over the parameters to this function via kwargs?